File: Un poco menos/Second/VerificaPalindromo.java


/*
 * Solicita al usuario una palabra o frase
 * Determina si la entrada es un palíndromo (igual al revés)
 * 
 * @author: Gonzalo Pulleiro
 */

import java.util.Scanner;

public class VerificaPalindromo {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Por favor, introduce una palabra o una frase...");

        String entrada = sc.nextLine();
        sc.close();

        entrada = entrada.toLowerCase().replaceAll("[^a-z0-9]", "");

        String salida = new StringBuilder(entrada).reverse().toString();

        System.out.println("Introducido: " + entrada + ". Obtenido: "+salida);

        if(entrada.equals(salida)){
            System.out.println("Palíndromo");
        }
        else {
            System.out.println("NO Palíndromo");
        }

    }
    
}

File: Un poco menos/Second/VerificaPalindromo.py
'''
 Solicita al usuario una palabra o frase
 Determina si la entrada es un palíndromo (igual al revés)
 
 @author: Gonzalo Pulleiro
'''
import re
import unicodedata

entrada = input("Por favor, introduce una palabra o una frase...\n")

entrada = unicodedata.normalize('NFD', entrada)
entrada = ''.join(c for c in entrada if unicodedata.category(c) != 'Mn')

entrada = re.sub(r'[^a-z0-9]', '', entrada.lower())

salida = entrada[::-1]

print("Inroducido: {}. Obtenido: {}".format(entrada, salida))

if entrada == salida:
    print("Palíndromo.")
else:
    print("NO Palíndromo")